haystack/components/embedders/azure_document_embedder.py

```diff
@@ -0,0 +1,182 @@
+import os
+from typing import List, Optional, Dict, Any, Tuple
+
+from openai.lib.azure import AzureADTokenProvider, AzureOpenAI
+from tqdm import tqdm
+
+from haystack import component, Document, default_to_dict
+
+
+@component
+class AzureOpenAIDocumentEmbedder:
+    """
+    A component for computing Document embeddings using OpenAI models.
+    The embedding of each Document is stored in the `embedding` field of the Document.
+
+    Usage example:
+    ```python
+    from haystack import Document
+    from haystack.components.embedders import AzureOpenAIDocumentEmbedder
+
+    doc = Document(content="I love pizza!")
+
+    document_embedder = AzureOpenAIDocumentEmbedder()
+
+    result = document_embedder.run([doc])
+    print(result['documents'][0].embedding)
+
+    # [0.017020374536514282, -0.023255806416273117, ...]
+    ```
+    """
+
+    def __init__(
+        self,
+        azure_endpoint: Optional[str] = None,
+        api_version: Optional[str] = "2023-05-15",
+        azure_deployment: Optional[str] = None,
```

Comment:
```suggestion
        azure_deployment: str = "text-embedding-ada-002",
```

Can it make sense to provide a default value here?

Reply:
O wow, good catch. I thought I did that already

Reply:
Aha I went for `self.azure_deployment = azure_deployment or "text-embedding-ada-002"` what do we do in these situations in other components?

Reply:
Ok, deal, I remember - will change then